File: Selenium/Project/Activity9.java
package HRM;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.BeforeMethod;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.annotations.AfterTest;
import org.testng.annotations.Test;

import java.util.List;

public class Activity9 {
    WebDriver driver;

    @BeforeMethod
    public void setUp() {
        WebDriverManager.chromedriver().setup();
        //Create a new instance of the Firefox driver
        driver = new ChromeDriver();
        // Open the URL
        driver.get("http://alchemy.hguy.co/orangehrm");
    }
    @Test
    public void retrieveEmergencyContacts() {
        // Login with credentials
        driver.findElement(By.id("txtUsername")).sendKeys("orange");
        driver.findElement(By.id("txtPassword")).sendKeys("orangepassword123");
        driver.findElement(By.id("btnLogin")).click();

        // Navigate to "My Info" using different selectors
        driver.findElement(By.linkText("My Info")).click(); // Using Link Text
        // Click on "Emergency Contacts"
        driver.findElement(By.linkText("Emergency Contacts")).click();

        // Retrieve and print information about all contacts listed in the table
        List<WebElement> contactRows = driver.findElements(By.cssSelector("table#emgcontact_list tbody tr"));
        List<String> contactInfo;

        for (int i = 0; i < contactRows.size(); i++) {
            WebElement row = contactRows.get(i);
            List<WebElement> columns = row.findElements(By.tagName("td"));

            String contactName = columns.get(1).getText();
            String contactRelationship = columns.get(2).getText();
            String contactHomePhone = columns.get(3).getText();
            String contactMobilePhone = columns.get(4).getText();
            String contactWorkPhone = columns.get(5).getText();

            contactInfo = List.of(
                    "Name: " + contactName,
                    "Relationship: " + contactRelationship,
                    "Home Phone: " + contactHomePhone,
                    "Mobile Phone: " + contactMobilePhone,
                    "Work Phone: " + contactWorkPhone
            );

            System.out.println("Contact Information:");
            for (String info : contactInfo) {
                System.out.println(info);
            }
            System.out.println();
        }
    }

    @AfterTest
    public void tearDown() {
        driver.quit();
    }
}

File: Selenium/Project/Activity9.py
from  selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.service import Service as ChromeService
from selenium.webdriver.support.select import Select
from webdriver_manager.chrome import ChromeDriverManager
driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))
from selenium.webdriver.common.keys import Keys

# Open the OrangeHRM page
url = "http://alchemy.hguy.co/orangehrm"
driver.get(url)

# Find and fill in the username and password fields
username = driver.find_element(By.ID, "txtUsername")
password = driver.find_element(By.ID, "txtPassword")
username.send_keys("orange")
password.send_keys("orangepassword123")
password.send_keys(Keys.RETURN)

# Navigate to the "My Info" page
menu_my_info = driver.find_element(By.ID, "menu_pim_viewMyDetails")
menu_my_info.click()

# Locate the left-hand menu and click on "Emergency Contacts"
menu_emergency_contacts = driver.find_element(By.ID, "sidenav")
menu_emergency_contacts.find_element(By.LINK_TEXT, "Emergency Contacts").click()

# Retrieve information about all the contacts listed in the table
contacts_table = driver.find_element(By.ID, "emgcontact_list")
rows = contacts_table.find_elements(By.TAG_NAME, "tr")

contact_info = []

# Loop through the table rows to collect contact information
for row in rows[1:]:  # Skip the header row
    columns = row.find_elements(By.TAG_NAME, "td")
    contact_name = columns[1].text
    contact_relationship = columns[2].text
    contact_home_phone = columns[3].text
    contact_mobile_phone = columns[4].text
    contact_work_phone = columns[5].text

    contact_info.append(
        {
            "Name": contact_name,
            "Relationship": contact_relationship,
            "Home Phone": contact_home_phone,
            "Mobile Phone": contact_mobile_phone,
            "Work Phone": contact_work_phone,

        }
    )

# Print all the information to the console
for contact in contact_info:
    print("Contact Information:")
    for key, value in contact.items():
        print(f"{key}: {value}")
    print()

# Close the browser
driver.quit()
